versioned_docs/version-2021sp/lecture5.md

```diff
@@ -3,8 +3,211 @@ id: lecture5
 title: Lecture 5
 ---
 
-Lecture Video will be released after lecture.
+[Lecture Video](https://drive.google.com/file/d/1az7P1eYb42oSr8aF1-30lCED9WoRWyVh/view?usp=sharing)
 
 [Lecture Slides](https://docs.google.com/presentation/d/1EkK8zboXxakJYecXWlGR93NvyTevfyy6byxpXZEQVgI/edit?usp=sharing)
 
 [Assignment 4](/docs/assignment4) (due 4/1 3:59 PM on CMS. This is not an April Fool's joke, we swear.)
+
+## Your First Component
+
+```tsx title="ContactCard.tsx"
+import React from 'react';
+
+type Props = { readonly name: string; readonly githubLink: string };
+
+const ContactCard = ({ name, githubLink }: Props) => (
+  <div>
+    <div>
+      You can reach to {name} at
+      <a href={githubLink}>{githubLink}</a>
+    </div>
+  </div>
+);
+```
+
+## Wat! What's going on
+
+### Functional Component
+
+The simplest component in React is a functional component. A functional component
+does not have any internal state. You can think of it as a function whose inputs
+are some JavaScript object and the output is some HTML code that is generated
+from the data.
+
+In React, we call the JavaScript object `props`, so you can see code like this:
+
+```tsx title="ContactCard.tsx"
+import React from 'react';
+
+type Props = { readonly name: string; readonly githubLink: string };
+
+const ContactCard = (props: Props) => (
+  <div>
+    <div>
+      You can reach to {props.name} at
+      <a href={props.githubLink}>{props.githubLink}</a>
+    </div>
+  </div>
+);
+```
+
+Just calling the input `props` is not good for documentation purpose, so we
+usually use the object destruct syntax to make it more explicit:
```

Comment:
Lets go with "object destructuring" just to be consistent with how we've explained it before

Reply:
gotcha